### ISSUE TITLE
Can't format code using Coc (without elm project)

### ISSUE THREAD

**Issue by netikular** [elm-tooling/elm-language-server#163]
While setting up vim with the Coc plugin the code formatting feature is not working.

## Expected Behavior
when using:
:call CocActionAsync('format')
The code should be formatted. 

## Current Behavior
When using:
:call CocActionAsync('format')
Nothing happens.

## Steps to Reproduce (for bugs)

1. Create a simple elm file
Main.elm:
```
module Main exposing (..)
add =
  1 + 1
```
2. Load file in vim
3. Try to format

## Context
Just trying to get Coc configured so that I can use the elm-language-server in vim.  I have to manually format my elm code.

## Your Environment
vim version: NVIM v0.4.2
node version: v8.11.3
coc.nvim version: 0.0.74-e04013a8bd
elm version: 0.19.0
MacOS 10.14.6

**Comment by netikular** [comment]
Hmm, I was able to get the formatting to work if I highlighted the text to format.

However the LS does not format the same as the elm-format command.  Is that expected?

**Comment by Razzeee** [comment]
Can you be more specific about what's different? We're calling `elm-format` in the backend and getting the result from there. But we do some things to make it nicer, so I need more context to help :)

Not a vim user, but I would expect format to work without highlighting. Hopefully @andys8 can help you.

**Comment by andys8** [comment]
I'm using the same command for formatting and it's working: https://github.com/andys8/dotfiles/blob/41d9f97be80b3819e2bd5c07a6f52b1ffae845cc/vimrc.local#L208

Can you provide more details (configuration or logs (`:CocInfo` and verbose logging))?


Or try `:call CocAction('format')`. Maybe if there is a timeout or an error, synchronous formatting behaves behaves differently and shows what's wrong.

**Comment by ni-ko-o-kin** [comment]
Try runnig `:CocConfig` and then paste and save this:

```
{
  "coc.preferences.formatOnSaveFiletypes": ["elm"],
  "languageserver": {
    "elmLS": {
      "command": "elm-language-server",
      "filetypes": ["elm"],
      "rootPatterns": ["elm.json"],
      "initializationOptions": {
        "elmAnalyseTrigger": "change"
       }
     }
  }
}
```

**Comment by netikular** [comment]
With the config that I had for Coc and using the config that was provided by @ni-ko-o-kin I get the following when using `:call CocActionAsnyc('format')`.

```
[Info  - 15:42:20] Done parsing all files.
[Info  - 15:42:25] Formatting was requested
[Warn  - 15:42:25] {}
```

The formatting was two space indent and only 2 lines after module definition, but today that is not even working.  So that might have been a red-herring.

**Comment by Razzeee** [comment]
Did you try `:call CocAction('format')`?

**Comment by netikular** [comment]
I did, same result.

**Comment by andys8** [comment]
I re-read the issue and tested what you described. I think the interesting part is: There is a single `Main.elm` file but without a project. And this actually is not working for me; in any editor.

Also it's not about formatting only. In vs code it looks like the language server won't be starting at all.

In vim the server is started, but formatting will fail with:

```
[Trace - 12:44:33 AM] Received response 'textDocument/formatting - (1)' in 2ms. Request failed: Unhandled method textDocument/formatting (-32601).
[Error  - 12:44:33 AM] Request textDocument/formatting failed.
  Message: Unhandled method textDocument/formatting
  Code: -32601 
```

**Comment by netikular** [comment]
I had run Elm init in the directory where the Main.elm file is, otherwise I noticed the Language Server would not start either, I actually saw a parse error.

**Comment by andys8** [comment]
So your issue is still about formatting only? Other features like diagnostics are working?

Maybe let's use https://github.com/rtfeldman/elm-spa-example as reference.

Could you please provide:

- the output of `which elm elm-format elm-test`
- the vim config
- the Coc config
- the steps you do to execute formatting

**Comment by netikular** [comment]
Nothing is working, 

```
elm-spa-example (master) > which elm elm-format elm-test
/Users/kevin/.asdf/shims/elm
/Users/kevin/.asdf/shims/elm-format
/Users/kevin/.asdf/shims/elm-test
```

Vim config is at: https://github.com/netikular/dotfiles/blob/master/vimrc

```
{
  "coc.preferences.formatOnSaveFiletypes": ["elm"],
  "languageserver": {
    "elmLS": {
      "command": "elm-language-server",
      "filetypes": ["elm"],
      "rootPatterns": ["elm.json"],
      "initializationOptions": {
        "elmAnalyseTrigger": "change"
       }
     }
  }
}
```

I was just saving the file expecting it to format.  Also I was using the `:call CocActionAsnyc('format')` on a file.

I had thought that I had created an elm project - and it was not working.  I've tried using the elm-spa-example and everything seems to be working, so I went and created a new elm project and it all seems to be working now.

I think that I'm going to chalk it up to a PEBKAC problem.  There seems to be something around not having the .elm file in the src folder and new files being created.

I went back to my previous project and moved the Main.elm file into the src/ dir and restarted vim and everything seems to be working now.

Thanks for all the help.
